src/plugin/PluginManager.php

```diff
@@ -653,8 +677,25 @@ public function registerEvent(string $event, \Closure $handler, int $priority, P
 
 		$timings = Timings::getEventHandlerTimings($event, $handlerName, $plugin->getDescription()->getFullName());
 
-		$registeredListener = new RegisteredListener($handler, $priority, $plugin, $handleCancelled, $timings);
+		if(is_subclass_of($event, AsyncEvent::class) && $this->canHandleAsyncEvent($handler)){
```

Comment:
`registerEvent` is a low-level API that does not perform any reflection checks on `$handler`. It doesn't look right to start introducing magical behavior to `registerEvent`; that's the job of `registerEvents`.